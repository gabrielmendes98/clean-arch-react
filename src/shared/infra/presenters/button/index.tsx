type Props = {
  children: React.ReactNode;
  className?: string;
  type?: 'button' | 'submit' | 'reset';
};

export const Button = ({
  children,
  className = '',
  type = 'button',
}: Props) => (
  <button className={className} type={type}>
    {children}
  </button>
);
