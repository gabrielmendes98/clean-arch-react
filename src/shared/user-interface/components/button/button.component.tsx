import { memo, MouseEvent } from 'react';

type Props = {
  children: React.ReactNode;
  className?: string;
  type?: 'button' | 'submit' | 'reset';
  onClick?: (e: MouseEvent<HTMLButtonElement, globalThis.MouseEvent>) => void;
};

const ButtonComponent = ({
  children,
  className = '',
  type = 'button',
  onClick,
}: Props) => (
  <button className={className} type={type} onClick={onClick}>
    {children}
  </button>
);

export const Button = memo(ButtonComponent);
