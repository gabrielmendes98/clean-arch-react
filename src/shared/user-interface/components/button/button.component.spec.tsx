import { render, screen, userEvent } from 'shared/testing/test-utils';
import { Button } from './button.component';

describe('Button', () => {
  it('should call onClick', () => {
    const onClick = jest.fn();
    render(<Button onClick={onClick}>click</Button>);
    userEvent.click(screen.getByRole('button', { name: 'click' }));
    expect(onClick).toHaveBeenCalled();
  });

  it('should match snapshot', () => {
    const { asFragment } = render(<Button>Some text</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
});
