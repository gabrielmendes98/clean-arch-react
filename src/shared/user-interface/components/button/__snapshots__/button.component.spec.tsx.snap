// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
<DocumentFragment>
  <button
    class=""
    type="button"
  >
    Some text
  </button>
</DocumentFragment>
`;
